Tests/UIKitTests/UIImageExtensionsTests.swift

```diff
@@ -13,6 +13,40 @@ import XCTest
 import UIKit
 
 final class UIImageExtensionsTests: XCTestCase {
+    func testAverageColor() {
+        // note that not all colors precisely survive the roundtrip due to colorspace
+        // management. Use XCTAssertEqual w/ accuracy.
+        func assertColorsEqual(_ color1: UIColor, _ color2: UIColor, accuracy: CGFloat = 0.01) {
+            var r1: CGFloat = 0, g1: CGFloat = 0, b1: CGFloat = 0, a1: CGFloat = 0
+            var r2: CGFloat = 0, g2: CGFloat = 0, b2: CGFloat = 0, a2: CGFloat = 0
+            color1.getRed(&r1, green: &g1, blue: &b1, alpha: &a1)
+            color2.getRed(&r2, green: &g2, blue: &b2, alpha: &a2)
```

Comment:
In tests we don't have the same limitation for not being able to use other extensions. In `UIColor` we have `rgbComponents` and `alpha`, so that can simplify things for you. E.g.
```Swift
let (r1, g1, b1) = color1.rgbComponents
let (r2, g2, b2) = color2.rgbComponents
```

Reply:
I experimented a bit. I think `rgbComponents` doesn't work quite as well because it returns Ints, which makes it more difficult to use XCTAssertEqual with `accuracy`. I'd have to cast to floats every step of the way. Good idea, though.